common/beerocks/bwl/dwpal/base_wlan_hal_dwpal.cpp

```diff
@@ -507,6 +552,154 @@ bool base_wlan_hal_dwpal::attach_ctrl_interface(int vap_id)
     return true;
 }
 
+bool base_wlan_hal_dwpal::process_nl_events()
+{
+    // process all NL events
+    if (!m_dwpal_nl_ctx) {
+        LOG(ERROR) << "Invalid WPA Control socket used for nl events (m_dwpal_ctx == nullptr)";
+        return false;
+    }
+
+    // Passing a lambda with capture is not supported for standard C function
+    // pointers. As a workaround, we create a static (but thread local) wrapper
+    // function that calls the capturing lambda function.
+    static __thread std::function<DWPAL_Ret(struct nl_msg * msg)> nl_handler_cb_wrapper;
+    nl_handler_cb_wrapper = [&](struct nl_msg *msg) -> DWPAL_Ret {
+        if (process_dwpal_nl_event(msg) != true) {
+            LOG(ERROR) << "User's netlink handler function failed!";
+            return DWPAL_FAILURE;
+        }
+        return DWPAL_SUCCESS;
+    };
+    auto nl_handler_cb = [](struct nl_msg *msg) -> DWPAL_Ret { return nl_handler_cb_wrapper(msg); };
+
+    if (m_fd_nl_events > 0) {
+        //parsing will be done in callback function
+        if (dwpal_driver_nl_msg_get(m_dwpal_nl_ctx, DWPAL_NL_UNSOLICITED_EVENT, NULL,
```

Comment:
All the code above seems to be quite a lot of complexity for essentially wrapping this one function call. Can't we just override process_nl_events() in the leaf class directly?